OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -40,7 +40,17 @@ public void Created(Actor self)
 
 		public IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self)
 		{
-			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
+			var player = self.World.RenderPlayer ?? self.World.LocalPlayer;
+			var color = self.Owner.Color.RGB;
+
+			if (Game.Settings.Game.AllianceDisplay && !self.Owner.NonCombatant)
+			{
+				color = self.Owner == player ? Color.LimeGreen :
+					self.Owner.IsAlliedWith(player) ? Color.Yellow : Color.Red;
```

Comment:
Agreed, may want to use similar logic to the health bar coloring if it's meant to behave exact: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Graphics/SelectionBarsRenderable.cs#L77